### PR TITLE
Uncompress streams without having to download to tmp first

### PR DESCRIPTION
**Pull request by niketh**
Uncompress streams without having to download to tmp first

### CODE REVIEW

File: extensions-core/protobuf-extensions/dependency-reduced-pom.xml [hunk 1]
```diff
@@ -0,0 +1,210 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (b-slim) [inline extensions-core/protobuf-extensions/dependency-reduced-pom.xml]: is this change needed ?

**Comment by b-slim** [comment]
@niketh how about the HDFS Puller ? also curious what is the main driver is it perf ? stability ? misc ?

File: extensions-core/protobuf-extensions/dependency-reduced-pom.xml (hunk 1 above)
Comment (niketh) [inline extensions-core/protobuf-extensions/dependency-reduced-pom.xml]: No required, removing it

**Comment by niketh** [comment]
@b-slim HDFS puller already uses directly used compressing from stream. This PR is a result of https://github.com/druid-io/druid/pull/4302

**Comment by b-slim** [comment]
ok looks like it is a performance improvement plus making pullers more uniform. 👍

**Comment by fjy** [comment]
👍
